test/IntentCompressor.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers'
+import { expect } from 'chai'
+import { ethers } from 'hardhat'
+import {
+  TestERC20,
+  IntentSource,
+  TestProver,
+  Inbox,
+  IntentCompressor,
+} from '../typechain-types'
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers'
+import { EncodedFulfillmentStruct, EncodedIntentStruct } from '../typechain-types/contracts/compressor/IntentCompressor'
+
+describe('Intent Compressor Test', (): void => {
```

Comment:
Should we start foundry tests? 😊 
So we don't have to rewrite it later when we migrate?

Reply:
Maybe starting from the next PRs since you already wrote all the tests